hasher-matcher-actioner/hmalib/common/timebucketizer.py

```diff
@@ -174,8 +178,53 @@ def get_records(
 
         for file in file_list:
             with open(file, "r") as my_file:
-                content_list.extend(
-                    list(map(type_class().from_csv, csv.reader(my_file)))
-                )
+                content_list.extend(list(map(type_class.from_csv, csv.reader(my_file))))
 
         return content_list
+
+    @staticmethod
+    def squash_content(
+        location: datetime.datetime,
+        type: str,
+        storage_path: str,
+        bucket_width: datetime.timedelta,
+        date_to: datetime.timedelta = datetime.timedelta(days=1),
+    ):
+
+        if bucket_width > date_to:
+            return
+
+        until_nearest = TimeBucketizer._calculate_bucket_endpoints(
+            location, bucket_width
+        )[1]
+
+        since_nearest = TimeBucketizer._calculate_bucket_endpoints(
+            until_nearest - date_to, bucket_width
+        )[0]
+
+        while until_nearest > since_nearest:
+
+            directory_path = TimeBucketizer._generate_path(
+                storage_path, type, until_nearest
+            )
+
+            if os.path.isdir(directory_path):
+                file_list = []
+                for file in os.listdir(directory_path):
+                    file_path = os.path.join(directory_path, file)
+                    if file.startswith("squash"):
+                        return
+                    elif os.path.isfile(file_path):
+                        file_list.append(file_path)
+
+                with open(
+                    os.path.join(directory_path, "squash" + str(uuid.uuid1())) + ".csv",
+                    "w",
+                ) as new_file:
+                    for file in file_list:
+                        with open(file, "r") as reader:
+                            new_file.write(reader.read())
+
+                        os.remove(file)
```

Comment:
Good candidate for an internal private style method.

There are two parts to this algorithm (identify, execute). If you pull out "execute" into a different method, in the future, we could define other "identify" algorithms while relying on the robust "execute" method you wrote.